src/codeu/chat/server/JSON.java

```diff
@@ -267,11 +298,29 @@ public Model readFromFile (String file) throws IOException
                             jp.nextToken();
                             time = Time.fromMs(jp.getLongValue());
                         }
-                        if (!(name.equals("")) && time != null && id != null) {
-                            model.add(new User(id, name, time));
+                        if (jp.getText().equals("lastUpdateTime")) {
+                            jp.nextToken();
+                            updateTime = jp.getLongValue();
+                        }
+                        if (!(name.equals("")) && time != null && id != null && userInterestsRun && conversationInterestsRun && updateTime!= -1) {
```

Comment:
In the real world we probably would have wanted to preserve backwards compatibility for old JSON files, so we wouldn't have required the interest tags, just supported them if they were there. I would say let's not worry about that right now because we have full control of the server and the JSON file, just something to think about.